diff --git a/lib/fetch.d.ts b/lib/fetch.d.ts
index 77ead6cdb884b5887ad22a1972ef46822cc005bf..300c0cea1aff9d3016ef55975cbaa86985abeba2 100644
--- a/lib/fetch.d.ts
+++ b/lib/fetch.d.ts
@@ -1,5 +1,7 @@
 import { type RetryTimeoutOptions } from '@zkochan/retry';
+// @ts-expect-error
 import { type Request, type RequestInit as NodeRequestInit, Response } from 'node-fetch';
+// @ts-expect-error
 export { isRedirect } from 'node-fetch';
 export { Response, type RetryTimeoutOptions };
 interface URLLike {
