diff --git a/lib/platform/node.js b/lib/platform/node.js
index 458329ea1e4b256df51e47433c9913a44360d68d..9dda097f2c11efdb10c3a89d018f47022e30d00b 100644
--- a/lib/platform/node.js
+++ b/lib/platform/node.js
@@ -4,13 +4,32 @@ Object.defineProperty(exports, '__esModule', { value: true });
 
 var tty = require('tty');
 
-function _interopDefaultLegacy (e) { return e && typeof e === 'object' && 'default' in e ? e : { 'default': e }; }
+function _interopNamespace(e) {
+    if (e && e.__esModule) return e;
+    var n = Object.create(null);
+    if (e) {
+        Object.keys(e).forEach(function (k) {
+            if (k !== 'default') {
+                var d = Object.getOwnPropertyDescriptor(e, k);
+                Object.defineProperty(n, k, d.get ? d : {
+                    enumerable: true,
+                    get: function () {
+                        return e[k];
+                    }
+                });
+            }
+        });
+    }
+    n['default'] = e;
+    return Object.freeze(n);
+}
 
-var tty__default = /*#__PURE__*/_interopDefaultLegacy(tty);
+var tty__namespace = /*#__PURE__*/_interopNamespace(tty);
 
 function getDefaultColorDepth() {
-    if (tty__default['default'] && `getColorDepth` in tty__default['default'].WriteStream.prototype)
-        return tty__default['default'].WriteStream.prototype.getColorDepth();
+    var _a, _b;
+    if (typeof ((_b = (_a = tty__namespace === null || tty__namespace === void 0 ? void 0 : tty.WriteStream) === null || _a === void 0 ? void 0 : _a.prototype) === null || _b === void 0 ? void 0 : _b.getColorDepth) === 'function')
+        return tty.WriteStream.prototype.getColorDepth();
     if (process.env.FORCE_COLOR === `0`)
         return 1;
     if (process.env.FORCE_COLOR === `1`)
