diff --git a/dist/src/index.d.ts b/dist/src/index.d.ts
index 409b9e4888cca0cb2f912da409bea35ba87c70ce..a24b6a6110e553cbbcc793826d2a1871da8ee716 100644
--- a/dist/src/index.d.ts
+++ b/dist/src/index.d.ts
@@ -1,12 +1,12 @@
 import { JSONSchema4 } from 'json-schema';
-import { Options as $RefOptions } from '@bcherny/json-schema-ref-parser';
+import $RefParser from '@bcherny/json-schema-ref-parser';
 import { Options as PrettierOptions } from 'prettier';
 export { EnumJSONSchema, JSONSchema, NamedEnumJSONSchema, CustomTypeJSONSchema } from './types/JSONSchema';
 export interface Options {
     /**
      * [$RefParser](https://github.com/BigstickCarpet/json-schema-ref-parser) Options, used when resolving `$ref`s
      */
-    $refOptions: $RefOptions;
+    $refOptions: $RefParser.Options;
     /**
      * Default value for additionalProperties, when it is not explicitly set.
      */
